lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs

```diff
@@ -196,28 +197,103 @@ spec :: forall n.
     , PaymentAddress n IcarusKey
     ) => SpecWith Context
 spec = describe "SHELLEY_TRANSACTIONS" $ do
-    it "TRANS_MIN_UTXO_01 - I cannot spend less than minUTxOValue" $ \ctx -> runResourceT $ do
-      wSrc <- fixtureWallet ctx
-      wDest <- emptyWallet ctx
-
-      let amt = minUTxOValue (_mainEra ctx) - 1
-      addrs <- listAddresses @n ctx wDest
-      let destination = (addrs !! 1) ^. #id
-      let payload = Json [json|{
-              "payments": [{
-                  "address": #{destination},
-                  "amount": {
-                      "quantity": #{amt},
-                      "unit": "lovelace"
-                  }
-              }],
-              "passphrase": #{fixturePassphrase}
-          }|]
-
-      let ep = Link.createTransactionOld @'Shelley
-      r <- request @(ApiTransaction n) ctx (ep wSrc) Default payload
-      expectResponseCode HTTP.status403 r
-      expectErrorMessage errMsg403MinUTxOValue r
+
+    it "TRANS_MIN_UTXO_01 - I cannot spend less than minUTxOValue" $
+        \ctx -> runResourceT $ do
+
+        wSrc <- fixtureWallet ctx
+        wDest <- emptyWallet ctx
+
+        let amt = minUTxOValue (_mainEra ctx) - 1
+        addrs <- listAddresses @n ctx wDest
+        let destination = (addrs !! 1) ^. #id
+        let payload = Json [json|{
+                "payments": [{
+                    "address": #{destination},
+                    "amount": {
+                        "quantity": #{amt},
+                        "unit": "lovelace"
+                    }
+                }],
+                "passphrase": #{fixturePassphrase}
+            }|]
+
+        let ep = Link.createTransactionOld @'Shelley
+        r <- request @(ApiTransaction n) ctx (ep wSrc) Default payload
+        expectResponseCode HTTP.status403 r
+        expectErrorMessage errMsg403MinUTxOValue r
+
+    it "TRANS_MIN_UTXO_BABBAGE_PRECISION - I can spend exactly minUTxOValue" $
+        \ctx -> runResourceT $ do
+
+        -- Coin selection will out of caution overestimate the size of the ada
+        -- quantity in its calculation when assigning a new minUTxOValue. In
```

Comment:
I think it might be worth explaining what we mean by "out of caution".

Suggestion:

> When producing **_auto-generated outputs_**, the coin selection and migration algorithms both use a minimum ada quantity function whose result can always be safely increased. This allows these algorithms the flexibility to freely reassign excess ada to auto-generated outputs without the risk of producing invalid ada quantities.

> This means that auto-generated outputs are assigned an initial ada quantity `a1` that is strictly in excess of the minimum required quantity `a0`, where the excess `a1 - a0` is exactly equal to `17,240 lovelace`.

> However, when validating **_user-specified outputs_** with pre-defined ada quantities, the coin selection algorithm merely **_verifies_** that the pre-defined ada quantities are valid **_without_** adjusting them in any way.

> This test verifies that the wallet doesn't require the aforementioned excess for user-specified values with pre-defined ada quantities.

Reply:
071cce4d12b6cb37b75bd51015475a720f5ef4a6 -  what do you think?

> When producing auto-generated outputs, the coin selection and migration algorithms both use a minimum ada quantity function whose result can always be safely increased. This allows these algorithms the flexibility to freely reassign excess ada to auto-generated outputs without the risk of producing invalid ada quantities.

I don't think we need/want to give rational for this _here_. But I changed "will out of caution" to "may", as I believe even some "ideal", fix-point-aware solution would be allowed to overestimate.